### ISSUE TITLE
external storage: FAT32 errors apparently related to copying files

### ISSUE THREAD

**Issue by ben-en** [Outernet-Project/lighthouse-firmware#9]
As the title says, i believe these errors are related to copying files from internal storage to external storage. There also may be some hangups with the config of paths, or FSAL.
## Plugin behavior

Status lights blink quickly then turn off

In syslog the following was observed:
#### FAT32 - microSD

```
Nov  4 19:00:12 outernet user.warn kernel: [ 3004.431918@0] normal card in
Nov  4 19:00:13 outernet user.info kernel: [ 3004.927457@0] sd: mmc_rescan_try_freq: trying to init card at 400000 Hz
Nov  4 19:00:13 outernet user.info kernel: [ 3005.141471@0] sd: new high speed SDHC card at address e624, clock 50000000, 4-bit-bus-width
Nov  4 19:00:13 outernet user.info kernel: [ 3005.146713@0] mmcblk0: sd:e624 SU04G 3.69 GiB 
Nov  4 19:00:13 outernet user.info kernel: [ 3005.179855@0]  mmcblk0: p1
Nov  4 19:00:13 outernet user.err wptdm[1169]: request failed: 409
Nov  4 19:00:14 outernet user.debug hotplug.mmcblk0p1: Checking filesystem using /sbin/fsck.fat
Nov  4 19:00:15 outernet user.debug hotplug.mmcblk0p1: Mounting /dev/mmcblk0p1 -> /mnt/mmcblk0p1
Nov  4 19:00:15 outernet user.info hotplug.mmcblk0p1: /dev/mmcblk0p1 mounted at /mnt/mmcblk0p1
Nov  4 19:00:15 outernet user.debug hotplug.mmcblk0p1: Processing external storage
Nov  4 19:00:18 outernet user.debug hotplug.mmcblk0p1: External storage platform is: (none)
Nov  4 19:00:18 outernet user.debug hotplug.mmcblk0p1: Preparing for external storage use
Nov  4 19:00:18 outernet user.debug hotplug.mmcblk0p1: Stopping services
Nov  4 19:00:18 outernet user.debug hotplug.mmcblk0p1: Stoppnig S91librarian
Nov  4 19:00:18 outernet user.debug hotplug.mmcblk0p1: Stoppnig S90fsal
Nov  4 19:00:18 outernet user.info ondd[1164]: [carousel] completed: content/ff0884037aebf7e3a809bf68d602898e.zip
Nov  4 19:00:18 outernet user.debug hotplug.mmcblk0p1: Stoppnig S80ondd
Nov  4 19:00:18 outernet user.info kernel: [ 3010.438932@1] wetek-dvb dvb.8: stop_feed_dmx0
Nov  4 19:00:18 outernet user.info kernel: [ 3010.438959@1] wetek-dvb dvb.8: FLUSH FIFO0
Nov  4 19:00:18 outernet user.info kernel: [ 3010.438970@1] wetek-dvb dvb.8: FLUSH ok
Nov  4 19:00:21 outernet user.debug hotplug.mmcblk0p1: Services stopped
Nov  4 19:00:21 outernet user.debug hotplug.mmcblk0p1: Initializing external storage
Nov  4 19:00:21 outernet user.debug hotplug.mmcblk0p1: Writing platform file to external storage
Nov  4 19:00:21 outernet user.debug hotplug.mmcblk0p1: Copying content from internal storage to external
Nov  4 19:00:21 outernet user.debug hotplug.mmcblk0p1: Copying /mnt/data/downloads/* -> /mnt/mmcblk0p1
Nov  4 19:00:21 outernet user.warn hotplug.mmcblk0p1: There were errors during copying
Nov  4 19:00:21 outernet user.info hotplug.mmcblk0p1: Using /dev/mmcblk0p1 as external storage
Nov  4 19:00:22 outernet user.debug hotplug.mmcblk0p1: Restarting services
Nov  4 19:00:22 outernet user.info ondd[1287]: [main] v2.1.0
Nov  4 19:00:22 outernet user.info ondd[1287]: [main] tuning to 1179 Mhz 22000 KSym
Nov  4 19:00:22 outernet user.debug hotplug.mmcblk0p1: Services started
```
#### FAT32 - eHDD

```
Nov  4 19:46:59 outernet user.warn kernel: [  468.997347@1] Indeed it is in host mode hprt0 = 00021501
Nov  4 19:46:59 outernet user.info kernel: [  469.177324@1] usb 2-1: new high-speed USB device number 2 using dwc_otg
Nov  4 19:46:59 outernet user.warn kernel: [  469.178322@0] Indeed it is in host mode hprt0 = 00001101
Nov  4 19:46:59 outernet user.info kernel: [  469.406280@0] usb 2-1: New USB device found, idVendor=1058, idProduct=0820
Nov  4 19:46:59 outernet user.info kernel: [  469.406536@0] usb 2-1: New USB device strings: Mfr=1, Product=2, SerialNumber=5
Nov  4 19:46:59 outernet user.info kernel: [  469.406736@0] usb 2-1: Product: My Passport 0820
Nov  4 19:46:59 outernet user.info kernel: [  469.407729@0] usb 2-1: Manufacturer: Western Digital
Nov  4 19:46:59 outernet user.info kernel: [  469.407981@0] usb 2-1: SerialNumber: 575846314538344135333057
Nov  4 19:46:59 outernet user.info kernel: [  469.448446@0] usb-storage 2-1:1.0: USB Mass Storage device detected
Nov  4 19:46:59 outernet user.info kernel: [  469.460265@0] scsi0 : usb-storage 2-1:1.0
Nov  4 19:47:00 outernet user.notice kernel: [  470.461580@0] scsi 0:0:0:0: Direct-Access     WD       My Passport 0820 1012 PQ: 0 ANSI: 6
Nov  4 19:47:00 outernet user.notice kernel: [  470.485901@1] sd 0:0:0:0: [sda] Spinning up disk...
Nov  4 19:47:00 outernet user.notice kernel: [  470.495795@0] scsi 0:0:0:1: Enclosure         WD       SES Device       1012 PQ: 0 ANSI: 6
Nov  4 19:47:06 outernet user.warn kernel: [  471.507140@1] ......ready
Nov  4 19:47:06 outernet user.notice kernel: [  476.598452@1] sd 0:0:0:0: [sda] 1953458176 512-byte logical blocks: (1.00 TB/931 GiB)
Nov  4 19:47:06 outernet user.notice kernel: [  476.605661@1] sd 0:0:0:0: [sda] Write Protect is off
Nov  4 19:47:06 outernet user.debug kernel: [  476.605823@1] sd 0:0:0:0: [sda] Mode Sense: 47 00 10 08
Nov  4 19:47:06 outernet user.err kernel: [  476.611323@1] sd 0:0:0:0: [sda] No Caching mode page found
Nov  4 19:47:06 outernet user.err kernel: [  476.611453@1] sd 0:0:0:0: [sda] Assuming drive cache: write through
Nov  4 19:47:06 outernet user.err kernel: [  476.632783@1] sd 0:0:0:0: [sda] No Caching mode page found
Nov  4 19:47:06 outernet user.err kernel: [  476.632926@1] sd 0:0:0:0: [sda] Assuming drive cache: write through
Nov  4 19:47:06 outernet user.info kernel: [  476.694214@1]  sda: sda1
Nov  4 19:47:07 outernet user.err kernel: [  476.712155@1] sd 0:0:0:0: [sda] No Caching mode page found
Nov  4 19:47:07 outernet user.err kernel: [  476.712297@1] sd 0:0:0:0: [sda] Assuming drive cache: write through
Nov  4 19:47:07 outernet user.notice kernel: [  476.712418@1] sd 0:0:0:0: [sda] Attached SCSI disk
Nov  4 19:47:07 outernet user.debug hotplug.sda1: Checking filesystem using /sbin/fsck.fat
Nov  4 19:47:07 outernet user.debug hotplug.sda1: Mounting /dev/sda1 -> /mnt/sda1
Nov  4 19:47:07 outernet user.info hotplug.sda1: /dev/sda1 mounted at /mnt/sda1
Nov  4 19:47:07 outernet user.debug hotplug.sda1: Processing external storage
Nov  4 19:47:10 outernet user.debug hotplug.sda1: External storage platform is: (none)
Nov  4 19:47:10 outernet user.debug hotplug.sda1: Preparing for external storage use
Nov  4 19:47:10 outernet user.debug hotplug.sda1: Stopping services
Nov  4 19:47:10 outernet user.debug hotplug.sda1: Stoppnig S91librarian
Nov  4 19:47:10 outernet user.debug hotplug.sda1: Stoppnig S90fsal
Nov  4 19:47:11 outernet user.debug hotplug.sda1: Stoppnig S80ondd
Nov  4 19:47:11 outernet user.info kernel: [  480.772296@1] wetek-dvb dvb.8: stop_feed_dmx0
Nov  4 19:47:11 outernet user.info kernel: [  480.772321@1] wetek-dvb dvb.8: FLUSH FIFO0
Nov  4 19:47:11 outernet user.info kernel: [  480.772332@1] wetek-dvb dvb.8: FLUSH ok
Nov  4 19:47:14 outernet user.debug hotplug.sda1: Services stopped
Nov  4 19:47:14 outernet user.debug hotplug.sda1: Initializing external storage
Nov  4 19:47:14 outernet user.debug hotplug.sda1: Writing platform file to external storage
Nov  4 19:47:14 outernet user.debug hotplug.sda1: Copying content from internal storage to external
Nov  4 19:47:14 outernet user.debug hotplug.sda1: Copying /mnt/data/downloads/* -> /mnt/sda1
Nov  4 19:47:14 outernet user.warn hotplug.sda1: There were errors during copying
Nov  4 19:47:14 outernet user.info hotplug.sda1: Using /dev/sda1 as external storage
Nov  4 19:47:14 outernet user.debug hotplug.sda1: Restarting services
Nov  4 19:47:14 outernet user.info ondd[1257]: [main] v2.1.0
Nov  4 19:47:14 outernet user.info ondd[1257]: [main] tuning to 1179 Mhz 22000 KSym
Nov  4 19:47:14 outernet user.debug hotplug.sda1: Services started
```
### Final note

On refresh files view only showed a .platform file, none of the content from /mnt/data/Downloads had been transferred.
## Removal behavior

On removal of FAT32/NTFS drives several things happen:
- FSAL crashes
- /mnt/data/downloads/ is unmounted
- as can be seen in the following logs, unmounting of /mnt/data/downloads/ is not logged.
#### FAT32 - microSD

```
Nov  4 19:10:16 outernet user.warn kernel: [ 3608.594565@1] card out
Nov  4 19:10:16 outernet user.info kernel: [ 3608.600238@1] sd: card e624 removed
Nov  4 19:10:16 outernet user.err kernel: [ 3608.603270@1] Buffer I/O error on device mmcblk0p1, logical block 1
Nov  4 19:10:16 outernet user.warn kernel: [ 3608.603363@1] lost page write due to I/O error on mmcblk0p1
Nov  4 19:10:17 outernet user.debug hotplug.mmcblk0p1: Stopping services
Nov  4 19:10:17 outernet user.debug hotplug.mmcblk0p1: Stoppnig S91librarian
Nov  4 19:10:17 outernet user.debug hotplug.mmcblk0p1: Stoppnig S90fsal
Nov  4 19:10:17 outernet user.debug hotplug.mmcblk0p1: Stoppnig S80ondd
Nov  4 19:10:17 outernet user.info kernel: [ 3609.059093@1] wetek-dvb dvb.8: stop_feed_dmx0
Nov  4 19:10:17 outernet user.info kernel: [ 3609.059117@1] wetek-dvb dvb.8: FLUSH FIFO0
Nov  4 19:10:17 outernet user.info kernel: [ 3609.059127@1] wetek-dvb dvb.8: FLUSH ok
Nov  4 19:10:20 outernet user.debug hotplug.mmcblk0p1: Services stopped
Nov  4 19:10:20 outernet user.debug hotplug.mmcblk0p1: Restarting services
Nov  4 19:10:20 outernet user.err kernel: [ 3612.137590@0] FAT-fs (mmcblk0p1): unable to read boot sector to mark fs as dirty
Nov  4 19:10:20 outernet user.info ondd[1340]: [main] v2.1.0
Nov  4 19:10:20 outernet user.info ondd[1340]: [main] tuning to 1179 Mhz 22000 KSym
Nov  4 19:10:20 outernet user.debug hotplug.mmcblk0p1: Services started
```
#### FAT32 - eHDD

```
Nov  4 19:50:59 outernet user.warn kernel: [  709.126737@0] Disconnect cb-Host
Nov  4 19:50:59 outernet user.info kernel: [  709.337681@1] usb 2-1: USB disconnect, device number 2
Nov  4 19:50:59 outernet user.debug hotplug.sda1: Stopping services
Nov  4 19:50:59 outernet user.debug hotplug.sda1: Stoppnig S91librarian
Nov  4 19:50:59 outernet user.debug hotplug.sda1: Stoppnig S90fsal
Nov  4 19:50:59 outernet user.debug hotplug.sda1: Stoppnig S80ondd
Nov  4 19:50:59 outernet user.info kernel: [  709.678523@1] wetek-dvb dvb.8: stop_feed_dmx0
Nov  4 19:50:59 outernet user.info kernel: [  709.678548@1] wetek-dvb dvb.8: FLUSH FIFO0
Nov  4 19:50:59 outernet user.info kernel: [  709.678557@1] wetek-dvb dvb.8: FLUSH ok
Nov  4 19:51:02 outernet user.err wptdm[1169]: request failed: 409
Nov  4 19:51:02 outernet user.debug hotplug.sda1: Services stopped
Nov  4 19:51:03 outernet user.err kernel: [  712.762915@1] FAT-fs (sda1): unable to read boot sector to mark fs as dirty
Nov  4 19:51:03 outernet user.debug hotplug.sda1: Restarting services
Nov  4 19:51:03 outernet user.info ondd[1312]: [main] v2.1.0
Nov  4 19:51:03 outernet user.info ondd[1312]: [main] tuning to 1179 Mhz 22000 KSym
Nov  4 19:51:03 outernet user.debug hotplug.sda1: Services started
```
### Final notes:
- On refresh of the files view no files are listed, though files are visible as expected with `ls` in /mnt/data/downloads/
- Reboot is required to restore correct folders.

**Comment by foxbunny** [comment]
This should now be fixed (See #7)

**Comment by ben-en** [comment]
Resolved as of 08149db471c0eb545632bde2d09ba9b3010e578a
